File: 01-SRP/cpp/bad_design.cpp
#include <iostream>
#include <string>

struct Order {
    int id;
    std::string item;
};

class OrderService {
public:
    void createOrder(const Order& order) {
        validateOrder(order);
        calculatePrice(order);
        saveToDatabase(order);
        sendConfirmationEmail(order);
        logEvent(order);
    }

private:
    void validateOrder(const Order& order) {
        std::cout << "Validating order\n";
    }

    void calculatePrice(const Order& order) {
        std::cout << "Calculating price\n";
    }

    void saveToDatabase(const Order& order) {
        std::cout << "Saving order to database\n";
    }

    void sendConfirmationEmail(const Order& order) {
        std::cout << "Sending confirmation email\n";
    }

    void logEvent(const Order& order) {
        std::cout << "Logging order event\n";
    }
};

int main() {
    Order order{101, "Laptop"};

    OrderService service;
    service.createOrder(order);

    return 0;
}

File: 01-SRP/python/bad_design.py
class OrderService:
    def create_order(self, order):
        self.validate_order(order)
        self.calculate_price(order)
        self.save_to_database(order)
        self.send_confirmation_email(order)
        self.log_event(order)

    def validate_order(self, order):
        print("Validating order")

    def calculate_price(self, order):
        print("Calculating price")

    def save_to_database(self, order):
        print("Saving order to database")

    def send_confirmation_email(self, order):
        print("Sending confirmation email")

    def log_event(self, order):
        print("Logging order event")


if __name__ == "__main__":
    order = {
        "id": 101,
        "items": ["Laptop", "Mouse"]
    }

    service = OrderService()
    service.create_order(order)